### ISSUE TITLE
SDK fns: GetAssetManager().GetProjectFileUrl() and GetProjectFileByName() behave differently w/ export folders

### ISSUE THREAD

**Issue by MikalDev** [Scirra/Construct-3-bugs#6413]
GetAssetManager().GetProjectFileUrl() and GetProjectFileByName() behave differently w/ export folders enabled, one uses folder paths (runtime) the other ignores folder paths (editor).

Export folders is set. When a plug-in uses GetProjectFileByName() to fetch files during editor, it appears that the parameter to pass must not include the subfolder names. When the plug-in uses GetAssetManager().GetProjectFileUrl()  during runtime, it appears that the parameter to pass does require the subfolder names.

## Attach a .c3p
[ProjectFileTest.zip](https://github.com/Scirra/Construct-3-bugs/files/10471464/ProjectFileTest.zip)

## Steps to reproduce

Since this is a plug-in SDK issue, a plug-in is needed to run the test project. It is a paid plug-in that I developed, so I cannot share it publicly. However, I am happy to send the addon file or a download key via a private channel (email, DM, etc.) Please let me know how to proceed with that.

1. Install 3DObject addon
2. Open project
3. Note that the 3D file is under a sub-folder, and the gltf path in the plug-in does uses a sub-folder.
4. Run project

## Observed result

After loading project, there will be an alert that the path cannot be found and the model will not display.
However, after running, the model will load and display.

## Expected result

After loading, model should display.
After running model should display.

## More details

**Affected browsers/platforms:** 

**First affected release:** Using latest beta (r326)

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by MikalDev** [comment]
My current workaround is to strip subfolders when in editor, but this would not work if the file names were the same, but in different subfolders.

```javascript
            if (gltfPath.includes('/')) {
                gltfPath = gltfPath.split('/').pop();
            }
            gltfURI = await runtime.GetProjectFileByName(gltfPath);

**Comment by AshleyScirra** [comment]
Can you either send the addon to ashley@scirra.com or make a minimal new addon that demonstrates the issue?

I think the problem is no official Construct features try to load project files as strings in the context of the editor, and so this code path is basically not implemented.

**Comment by MikalDev** [comment]
Email sent, thanks!

**Comment by AshleyScirra** [comment]
So the editor `IProject.GetProjectFileByName()` method does indeed only match by the filename, ignoring the path. This probably has to stay like that for the time being for backwards compatibility, but I suspect we will have to deprecate it in the long run: Construct still enforces all filenames are unique even when they're in different folders for compatibility with methods like this, but if we ever want to lift that restriction, methods like that will have to be removed.

For the next beta I've added a new editor method `IProject.GetProjectFileByExportPath()` which accepts a full path like your addon uses, and also abides by the 'Export file structure' setting (i.e. in flat mode it will use a case-insensitive filename only with no path). This should mean that paths looked up in the editor with that method always correspond to the paths that can be fetched at runtime. I've tested this with your addon and it looks like it's working correctly, so hopefully that's all sorted.
